tests/fixtures_new.py

```diff
@@ -0,0 +1,240 @@
+import pytest
+from tests import helpers
+from ocs_ci.ocs import constants, ocp
+
+
+@pytest.fixture()
+def secret(request, interface_type):
+    """
+    Create a secret
+
+    Returns:
+        OCS: An instance of secret OCS
+
+    """
+    secrets = list()
+
+    def finalizer():
+        """
+        Delete the RBD secret
+        """
+        for secret in secrets:
+            secret.delete()
+            secret.ocp.wait_for_delete(secret.name)
+
+    request.addfinalizer(finalizer)
+    secrets.append(helpers.create_secret(interface_type=interface_type))
+
+    return secrets
+
+
+@pytest.fixture()
+def pool(request, interface):
+    """
+    Create a pool
+
+    Returns:
+
+    """
+    pools = list()
+
+    def finalizer():
+        """
+        Delete the Ceph block pool
+        """
+        if interface == constants.CEPHBLOCKPOOL:
+            for pool in pools:
+                pool.delete()
+                pool.ocp.wait_for_delete(pool.name)
+
+    request.addfinalizer(finalizer)
+
+    if interface == constants.CEPHBLOCKPOOL:
+        pools.append(helpers.create_ceph_block_pool())
+    elif interface == constants.CEPHFILESYSTEM:
+        pools.append(helpers.get_cephfs_data_pool_name())
+    return pools
+
+
+@pytest.fixture()
+def storageclass(request, secret, pool, interface_type):
+    """
+    Create storage class
+
+    Returns:
+        OCS: An instance of storageclass OCS
+
+    """
+    storageclasses = list()
+
+    def finalizer():
+        """
+        Delete the storage class
+
+        """
+        for sc in storageclasses:
+            sc.delete()
```

Comment:
What if storageclass got deleted from test intentionally? Than your attempt to delete here will fail, I think you have to check if the sc is not already deleted.

Reply:
If the storageclass was deleted in the test then here storageclass will be an empty list and nothing will happen

Reply:
Can you give me example of delete from test?

if you will have 
```python
def test_sc(storageclass):
    storageclass[0].delete()
    #this won't delete the object from list provided by fixture.
    # This case you have to also do del myl[0] or something like:
    my_sc = storageclass.pop(0)
    my_sc.delete()
```

Reply:
@petr-balogh @ebenahar this seems like something we would like the `delete()` call to handle instead of having to perform the check in all. utilities / fixtures. https://github.com/red-hat-storage/ocs-ci/issues/557